packages/components/src/autocomplete/types.ts

```diff
@@ -0,0 +1,212 @@
+	 * trigger and query text and returns a boolean indicating whether the
+	 * completer should be considered for that context.
+	 */
+	allowContext?: ( before: string, after: string ) => boolean;
+	/**
+	 * A function that takes an option and returns how the option should
+	 * be completed. By default, the result is a value to be inserted in the text.
+	 * However, a completer may explicitly declare how a completion should be
+	 * treated by returning an object with `action` and `value` properties. The
+	 * `action` declares what should be done with the `value`.
+	 */
+	getOptionCompletion?: (
+		option: TCompleterOption,
+		query: string
+	) => OptionCompletion | OptionCompletion[ 'value' ];
+	/**
+	 * A function that returns an array of items to be displayed in the
+	 * Autocomplete UI. These items have uniform shape and have been filtered by
+	 * `AutocompleterUIProps.filterValue`.
+	 */
+	useItems?: ( filterValue: string ) => [ Array< KeyedOption > ];
+	/**
+	 * Whether or not changes to the `filterValue` should be debounced.
+	 */
+	isDebounced?: boolean;
+	/**
+	 * A CSS class name to be applied to the completion menu.
+	 */
+	className?: string;
+};
+
+type ContentRef = MutableRefObject< HTMLElement | undefined >;
+
+export type AutocompleterUIProps = {
+	/**
+	 * The value to filter the options by.
+	 */
+	filterValue: string;
+	/**
+	 * An id unique to each instance of the component, used in the IDs of the
+	 * buttons generated for individual options.
+	 */
+	instanceId: number;
+	/**
+	 * The id of to be applied to the listbox of options.
+	 */
+	listBoxId: string | undefined;
+	/**
+	 * The class to apply to the wrapper element.
+	 */
+	className?: string;
+	/**
+	 * The index of the currently selected option.
+	 */
+	selectedIndex: number;
+	/**
+	 * A function to be called when the filterValue changes.
+	 */
+	onChangeOptions: ( items: Array< KeyedOption > ) => void;
+	/**
+	 * A function to be called when an option is selected.
+	 */
+	onSelect: ( option: KeyedOption ) => void; //should the param be an index?
+	/**
+	 * A function to be called when the completer is reset (e.g. when the user hits the escape key).
+	 */
+	onReset?: () => void;
+	/**
+	 * A function that defines the behavior of the completer when it is reset
+	 */
+	reset: ( event: Event ) => void;
+	/**
+	 * The rich text value object the autocomleter is being applied to.
+	 */
+	value: RichTextValue;
+	/**
+	 * A ref containing the editable element that will serve as the anchor for `Autocomplete`'s `Popover`.
+	 */
+	contentRef: ContentRef;
+};
+
+/**
+ * A debounced promise used retrieve completer options.
+ *
+ * @see `get-default-use-items.tsx`
+ */
+export type DebouncedPromise< T = void > = Promise< T > & {
+	canceled?: boolean;
+};
+
+/**
+ * When `@wordpress/rich-text` is fully typed, the following
+ * types should be moved to and imported from there
+ *
+ * @see @wordpress/rich-text/src/create.js
```

Comment:
I'm not sure about the format of this `@see` tag, since it uses a mix of `npm` package name (the `@` sign) and the actual folder path in the repo (`src/create.js`).

Perhaps we could just use the file's absolute path in the repo instead ? (ie. `/packages/rich-text/....)`